tools/purge_orphans.rb

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env ruby
+require "optimist"
+
+VALID_MODES = %w[count purge].freeze
+opts = Optimist.options do
+  synopsis "Purge orphaned rows from several tables."
+  usage "[options]"
+
+  opt :mode, "Mode (#{VALID_MODES.join(", ")})", :default => "count", :permitted => VALID_MODES
```

Comment:
I just realized that now that you're using permitted you don't need to output the values yourself.
```suggestion
  opt :mode, "Mode", :default => "count", :permitted => VALID_MODES
```

Reply:
pushed, updated description